app/App/Main/Account/Requests/TransactionRequest/AdjustFee/index.js

```diff
@@ -25,17 +45,156 @@ function maxFee (tx = { chainId: '' }) {
   return 50 * 1e18
 }
 
-//  <div className='txModuleTop'>
-// <div className={'txModuleTopData txModuleTopDataExpanded'}>
-  // <div className='transactionDataNotice'>{svg.octicon('issue-opened', { height: 26 })}</div>
-  // <div className='transactionDataLabel'>View Data</div>
-  // <div className='transactionDataIndicator' onMouseDown={() => this.copyData(req.data.data)}>
-    // {svg.octicon('clippy', { height: 20 })}
-  // </div>
-// </div> 
-// <div className='txModuleBody'>
-
-class TxFeeOverlay extends React.Component {
+const defaultLimiter = (val) => trimGwei(limitRange(val))
+
+const FeeOverlayInput = ({ initialValue, labelText, tabIndex, decimals, onReceiveValue, limiter = defaultLimiter }) => {
+  const [value, setValue] = useState(initialValue)
+  const [submitTimeout, setSubmitTimeout] = useState(0)
+  const processValue = (newValue) => {
+    const limitedValue = limiter(newValue)
+    onReceiveValue(limitedValue)
+    setValue(formatForInput(limitedValue, decimals))
+  }
+  const set = (newValueStr, newValueNum) => {
```

Comment:
what do these two functions do? names aren't too helpful and it's kind of confusing

Reply:
yeah will tweak after tests